attestation-agent/attester/src/lib.rs

```diff
@@ -74,6 +74,10 @@ pub trait Attester {
     ) -> Result<()> {
         bail!("Unimplemented")
     }
+
+    async fn check_init_data(&mut self, _init_data: Vec<u8>) -> Result<()> {
```

Comment:
Does this have to be `&mut self`? 

Can we pass the _init_data as reference `&[u8]`?

Reply:
You are right. We do not have to mark this as `mut` as we do not change any inside state. Also, let me add another commit to change other trait functions who does need to be `mut`.

`&[u8]` is a good choice. Let me fix. Also this conversion reminds me of old times XD.